widget/menu_item.go

```diff
@@ -392,33 +415,63 @@ func shortcutColor(th fyne.Theme) color.Color {
 	return color.NRGBA{R: uint8(r), G: uint8(g), B: uint8(b), A: uint8(a)}
 }
 
-func textsForShortcut(s fyne.KeyboardShortcut, th fyne.Theme) (texts []*canvas.Text) {
-	b := strings.Builder{}
+// textsForShortcut return texts for rendering a shortcut.
+func textsForShortcut(s fyne.KeyboardShortcut, th fyne.Theme) []*canvas.Text {
+	return textsForShortcutAndOS(s, th, runtime.GOOS)
+}
+
+// textsForShortcut return texts for rendering a shortcut for a specific OS.
+func textsForShortcutAndOS(s fyne.KeyboardShortcut, th fyne.Theme, os string) []*canvas.Text {
+	var b strings.Builder
 	mods := s.Mod()
-	if mods&fyne.KeyModifierControl != 0 {
-		b.WriteRune(runeModifierControl)
-	}
-	if mods&fyne.KeyModifierAlt != 0 {
-		b.WriteRune(runeModifierAlt)
-	}
-	if mods&fyne.KeyModifierShift != 0 {
-		b.WriteRune(runeModifierShift)
-	}
-	if mods&fyne.KeyModifierSuper != 0 {
-		b.WriteRune(runeModifierSuper)
-	}
-	r := keySymbols[s.Key()]
-	if r != 0 {
-		b.WriteRune(r)
-	}
 	shortColor := shortcutColor(th)
-	t := canvas.NewText(b.String(), shortColor)
-	t.TextStyle.Symbol = true
-	texts = append(texts, t)
-	if r == 0 {
-		text := canvas.NewText(string(s.Key()), shortColor)
-		text.TextStyle.Monospace = true
-		texts = append(texts, text)
-	}
-	return
+	texts := make([]*canvas.Text, 0)
+	switch os {
+	case "darwin":
+		if mods&fyne.KeyModifierControl != 0 {
+			b.WriteRune(runeModifierControlDarwin)
+		}
+		if mods&fyne.KeyModifierAlt != 0 {
+			b.WriteRune(runeModifierAltDarwin)
+		}
+		if mods&fyne.KeyModifierShift != 0 {
+			b.WriteRune(runeModifierShiftDarwin)
+		}
+		if mods&fyne.KeyModifierSuper != 0 {
+			b.WriteRune(runeModifierSuperDarwin)
+		}
+		r, hasSymbol := keySymbolsDarwin[s.Key()]
+		if hasSymbol {
+			b.WriteRune(r)
+		}
+		t := canvas.NewText(b.String(), shortColor)
+		t.TextStyle.Symbol = true
+		texts = append(texts, t)
+		if !hasSymbol {
+			t := canvas.NewText(string(s.Key()), shortColor)
+			t.TextStyle.Monospace = true
+			texts = append(texts, t)
+		}
+	default:
+		if mods&fyne.KeyModifierControl != 0 {
+			b.WriteString("Ctrl+")
+		}
+		if mods&fyne.KeyModifierAlt != 0 {
+			b.WriteString("Alt+")
+		}
+		if mods&fyne.KeyModifierShift != 0 {
+			b.WriteString("Shift+")
+		}
+		if mods&fyne.KeyModifierSuper != 0 {
+			b.WriteString("Super+")
+		}
```

Comment:
This seems like quite a bit of code duplication. Can you somehow move the darwin rune constants to a separate file and define the values specifically on darwin and have the other values in another file? This way we could have `runeModifierControlDarwin` return the darwin value on darwin and "Ctrl+" on other operating systems.

Reply:
If you look more closely you will see that the code for each case is actually quite different. The former adds rune, the later strings. The former can add two texts with different formatting, the later only one. 

My intention here was to minimize changes to the existing code. I would advise against mapping the shortcut text via platform dependent constants though. That would make this function harder to test and understand. But I am happy to look into combining the two key maps into one, which should allow me to reduce the lines of code quite a bit.

Reply:
There appear to be some flaky tests and I do not have the rights to re-run tests in this repo, but they are now passing in my branch.

Reply:
I'm sorry but there seems like there have been some communication mishaps as some information got lost in translation.

You didn't have to rework the code that much, I find it a lot harder to read now, sorry. What I was trying to say was that the four if statements were duplicated. What I meant was this:
- Move darwin rune constants to a `_darwin.go` file so they are used only when building for that architecture. Use single length strings and remove darwin specific naming.
- The other ctrl+ string etc. for other architectures can be put in another file that only is compiled in on non darwin desktop os. Constants have the same names as corresponding darwin ones.
- Move the four if statements out of the darwin/other-os checks so they happen before that and use the containers that switch depending on the os in use. The previous code you had in this function for checking operating system can be kept.
- One test runs on darwin the other for other architectures. There should already be existing code/files for this (as parts of the menus are darwin specific already), I think. 

Again, I'm sorry if this was unclear but it was what I was trying to hunt at with my previous messages.

Reply:
There are 19 platform dependent strings, not 4. Are you suggesting to map all of those via target specific code or just the modifiers?

As I tried to explain before, it is not only the strings that are different, but there also is different code per platform. Do you want me to put that platform specific logic in target specific files too or keep the current approach via runtime constant?

Would it not make much more sense to use one approach to handle all platform specific variations, not mix approaches?

Again, would it not be better to avoid the additional complexity that target specific code adds and keep everything together in one file (i.e. use the runtime constant approach for everything)?

With all due respect, but it appears to me that you are asking changes without having a complete understanding of the  code. So in order to avoid any more miscommunication, I would very much appreciate it if you could take the time to read through the complete code (and my above questions) and then revise your comments accordingly.

Reply:
> With all due respect, but it appears to me that you are asking changes without having a complete understanding of the code. 

In general this seems like an unhelpful thing to say - maintainers of this project generally have a good handle of the code. If in doubt try to assume that.

I do not understand where the miscommunication exists but this diff seems to be getting longer somehow.
Previously there were 3 consts that defined the runes used. And I understand that there are now 19 strings. But even so the code complexity should not have needed to increase.

We already had the OS specific (for darwin) code through build flags which I am sure @Jacalz was indicating could be kept.

Personally I would have split this into two commits - the first moving from the old rune system to new strings (where darwin are 1-length and others vary) which should have been an almost direct replacement. And then a second commit introducing the idea that the key symbol/name lookup could be set per-system as well.

It is possible that I have over simplified here, and if so please do share why that wouldn't be desirable - but it seems like it would be the minimal approach.

Reply:
> In general this seems like an unhelpful thing to say - maintainers of this project generally have a good handle of the code. If in doubt try to assume that.

I did not say that. I was referring to the code of my PR, not the existing code of the Fyne project.

I am sorry, but it is still not clear to me what approach you want me to implement. 
Please allow me summarize the issue as I see it:

The baseline is my initial commit, not the latest commit (That is what I understand @Jacalz  is referring to).

There are 4 kinds of things that have to be defined for each platform:
- 4 modifiers (string mapping)
- 15 keys (string mapping)
- text format for modifiers (symbol for mac, default for others)
- text format for keys (symbol or monospace for mac, default for others)

There are 2 approaches to define something platform dependent:

- A: target specific code / build tags
- B: runtime OS condition

There are 2 solutions being discussed:

1. I would suggest to use approach B for everything, because it is simpler and allows you to unit test everything. While with approach A you can only test the whole thing in platform tests (e.g. on Github).

1. @Jacalz latest request would use approach A for the 4 modifiers and approach B for the rest. I find it pretty ugly to mix both approaches in the same solution.

I would appreciate if you could confirm that 2. is really the solution you want me to implement.